plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBInternalDatabaseInformationProvider.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.dbeaver.DBException;
+
+/**
+ * Internal database information provider.
+ * Contains information about database used for internal needs.
+ */
+public interface DBInternalDatabaseInformationProvider {
+
+    /**
+     * Internal database info.
+     */
+    @NotNull
```

Comment:
We need separate info about driver name, URL, database name, database version.
DBInternalDatabaseInformationProvider  -> DBPConnectionInformation